pkg/config/config.go

```diff
@@ -40,112 +59,112 @@ const (
 var (
 	ComputeExecutionsStorePath = filepath.Join(ComputeStorePath, "executions.db")
 	OrchestratorJobStorePath   = filepath.Join(OrchestratorStorePath, "jobs.db")
-)
 
-var (
 	environmentVariableReplace = strings.NewReplacer(".", "_")
 	DecoderHook                = viper.DecodeHook(mapstructure.TextUnmarshallerHookFunc())
 )
 
-const (
-	ConfigFileName = "config.yaml"
-	ConfigFileMode = 0666
-)
-
-func Init(path string) (types.BacalhauConfig, error) {
-	// initialize the configuration with default values.
-	return initConfig(path,
-		WithDefaultConfig(getDefaultConfig(path)),
-	)
+type config struct {
+	// viper instance for holding user provided configuration
+	user *viper.Viper
+	// viper instance for holding system specific configuration
+	system *viper.Viper
+	// the default configuration values to initialize with
```

Comment:
I have to say this is very confusing for me to follow and understand. What I think we need to do in general is just use and pass `BacalhauConfig` around instead of `config.Context`. You can have different implementations to construct `BacalhauConfig` with viper being our only option for now. This allows you to test things without having to setup viper or have a file based config. Just construct `BacalhauConfig` instance, set some values, and pass it to your test.

What I think will simplify things is making the repo path part of `BacalhauConfig` instead of something independent. The [Rethinking Configuration](https://www.notion.so/expanso/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb) doc is suggesting that as well

Reply:
What this might look like is moving `types.BacalhauConfig` to `config.Config`, and moving the current viper implementation to `config/viper/...` as an implementation to setup our config

Reply:
The challenge that you might face is config setters. We need to decouple that from `BacalhauConfig` and we can have a separate interface for writing configs, with viper being an implementation.